README.rst

```diff
@@ -0,0 +1,3824 @@
+   INFO:hazelcast.lifecycle:HazelcastClient 4.0.0 is SHUTTING_DOWN
+   INFO:hazelcast.connection:Removed connection to Address(host=127.0.0.1, port=5701):7682c357-3bec-4841-b330-6f9ae0c08253, connection: Connection(id=0, live=False, remote_address=Address(host=172.17.0.2, port=5701))
+   INFO:hazelcast.lifecycle:HazelcastClient 4.0.0 is DISCONNECTED
+   INFO:hazelcast.lifecycle:HazelcastClient 4.0.0 is SHUTDOWN
+
+Congratulations. You just started a Hazelcast Python client.
+
+**Using a Map**
+
+Let’s manipulate a distributed map(similar to Python’s builtin ``dict``)
+on a cluster using the client.
+
+.. code:: python
+
+   import hazelcast
+
+   client = hazelcast.HazelcastClient()
+
+   personnel_map = client.get_map("personnel-map")
+   personnel_map.put("Alice", "IT")
+   personnel_map.put("Bob", "IT")
+   personnel_map.put("Clark", "IT")
+
+   print("Added IT personnel. Printing all known personnel")
+
+   for person, department in personnel_map.entry_set().result():
+       print("%s is in %s department" % (person, department))
+       
+   client.shutdown()
+
+**Output**
+
+::
+
+   Added IT personnel. Printing all known personnel
+   Alice is in IT department
+   Clark is in IT department
+   Bob is in IT department
+
+You see this example puts all the IT personnel into a cluster-wide
+``personnel-map`` and then prints all the known personnel.
+
+Now, run the following code.
+
+.. code:: python
+
+   import hazelcast
+
+   client = hazelcast.HazelcastClient()
+
+   personnel_map = client.get_map("personnel-map")
+   personnel_map.put("Denise", "Sales")
+   personnel_map.put("Erwing", "Sales")
+   personnel_map.put("Faith", "Sales")
+
+   print("Added Sales personnel. Printing all known personnel")
+
+   for person, department in personnel_map.entry_set().result():
+       print("%s is in %s department" % (person, department))
+
+   client.shutdown()
+
+**Output**
+
+::
+
+   Added Sales personnel. Printing all known personnel
+   Denise is in Sales department
+   Erwing is in Sales department
+   Faith is in Sales department
+   Alice is in IT department
+   Clark is in IT department
+   Bob is in IT department
+
+..
+
+   NOTE: For the sake of brevity we are going to omit boilerplate
+   parts, like ``import``\s, in the later code snippets. Refer to
+   the `Code Samples section <#16-code-samples>`__ to see samples
+   with the complete code.
+
+
+You will see this time we add only the sales employees but we get the
+list of all known employees including the ones in IT. That is because
+our map lives in the cluster and no matter which client we use, we can
+access the whole map.
+
+You may wonder why we have used ``result()`` method over the
+``entry_set()`` method of the ``personnel_map``. That is because the
+Hazelcast Python client is designed to be fully asynchronous. Every
+method call over distributed objects such as ``put()``, ``get()``,
+``entry_set()``, etc. will return a ``Future`` object that is similar to
```

Comment:
It would be good but we don't have a proper way to access the latest version of the API docs. So, If I do that, I could put a link to the 4.0 version (which is not released or created yet) but I am pretty sure that we will forget to update this link in upcoming releases. Since our version is very similar to the well-known `concurrent.futures.Future`, I think it should be okay as it is. What do you think?

Reply:
Sounds good to me.

Reply:
I mean keeping the text as is. And it would be certainly good to have a way to publish the latest API docs in the future.